torchmetrics/classification/accuracy.py

```diff
@@ -103,15 +152,31 @@ class Accuracy(Metric):
 
     def __init__(
         self,
+        num_classes: Optional[int] = None,
         threshold: float = 0.5,
+        average: str = "micro",
+        mdmc_average: Optional[str] = "global",
+        ignore_index: Optional[int] = None,
         top_k: Optional[int] = None,
+        multiclass: Optional[bool] = None,
         subset_accuracy: bool = False,
         compute_on_step: bool = True,
         dist_sync_on_step: bool = False,
         process_group: Optional[Any] = None,
         dist_sync_fn: Callable = None,
     ):
+        allowed_average = ["micro", "macro", "weighted", "samples", "none", None]
+        if average not in allowed_average:
+            raise ValueError(f"The `average` has to be one of {allowed_average}, got {average}.")
```

Comment:
please add what errors the classes raises to docstring, like:
https://github.com/PyTorchLightning/metrics/blob/b26c20d6f8b89bc03dd4ed8512828e38339d41af/torchmetrics/classification/precision_recall.py#L108-L110